core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -735,9 +735,9 @@ private[log] class Cleaner(val id: Int,
        *   1) if there exists a message with the same key but higher offset
        *   2) if the message is a delete "tombstone" marker and enough time has passed
        */
-      val redundant = foundOffset >= 0 && record.offset < foundOffset
+      val latest = record.offset() >= foundOffset
```

Comment:
I like this simplification but I think `latest` is still a confusing name since `map.latestOffset` is a separate thing. Maybe name it something like `latestOffsetForKey`?

Reply:
@bob-barrett  done